realm/realm-library/src/main/java/io/realm/internal/OsRealmConfig.java

```diff
@@ -0,0 +1,250 @@
+                    migrationCallback, initializationCallback);
+        }
+    }
+
+    private static final byte SCHEMA_MODE_VALUE_AUTOMATIC = 0;
+    private static final byte SCHEMA_MODE_VALUE_READONLY = 1;
+    private static final byte SCHEMA_MODE_VALUE_RESET_FILE = 2;
+    private static final byte SCHEMA_MODE_VALUE_ADDITIVE = 3;
+    private static final byte SCHEMA_MODE_VALUE_MANUAL = 4;
+
+    private final static long nativeFinalizerPtr = nativeGetFinalizerPtr();
+
+    private final RealmConfiguration realmConfiguration;
+    private final long nativePtr;
+    // Every SharedRealm instance has to be created from an OsRealmConfig instance. And the SharedRealm's NativeContext
+    // object will be the same as the context here. This is because of we may create different SharedRealm instances
+    // with different shared_ptrs which are point to the same SharedGroup object. It could happen when we create
+    // SharedRealm for migration/initialization callback. The context has to be the same object for those cases for
+    // core destructor's thread safety.
+    private final NativeContext context = new NativeContext();
+
+    private OsRealmConfig(final RealmConfiguration config,
+                          boolean autoUpdateNotification,
+                          OsSchemaInfo schemaInfo,
+                          SharedRealm.MigrationCallback migrationCallback,
+                          SharedRealm.InitializationCallback initializationCallback) {
+        this.realmConfiguration = config;
+        this.nativePtr = nativeCreate(config.getPath(), false, true);
+        NativeContext.dummyContext.addReference(this);
+
+        // Set encryption key
+        byte[] key = config.getEncryptionKey();
+        if (key != null) {
+            nativeSetEncryptionKey(nativePtr, key);
+        }
+
+        // Set durability
+        nativeSetInMemory(nativePtr, config.getDurability() == Durability.MEM_ONLY);
+
+        // Set auto update notification
+        nativeEnableChangeNotification(nativePtr, autoUpdateNotification);
+
+        // Sync config
+        Object[] syncUserConf = ObjectServerFacade.getSyncFacadeIfPossible().getUserAndServerUrl(realmConfiguration);
+        String syncUserIdentifier = (String) syncUserConf[0];
+        String syncRealmUrl = (String) syncUserConf[1];
+        String syncRealmAuthUrl = (String) syncUserConf[2];
+        String syncRefreshToken = (String) syncUserConf[3];
+        boolean syncClientValidateSsl = (Boolean.TRUE.equals(syncUserConf[4]));
+        String syncSslTrustCertificatePath = (String) syncUserConf[5];
+        if (syncRealmUrl != null) {
+            nativeCreateAndSetSyncConfig(nativePtr, syncRealmUrl, syncRealmAuthUrl, syncUserIdentifier,
+                    syncRefreshToken);
+            // FIXME: shouldn't we throw earlier for this case?
+            if (syncClientValidateSsl && syncSslTrustCertificatePath != null) {
```

Comment:
Yes. btw we do log a warning when the user disable the SSL validation and provide a `rootCA` path https://github.com/realm/realm-java/blob/930e1d00cac47149733d17df8688d49fb983d5ac/realm/realm-library/src/objectServer/java/io/realm/SyncConfiguration.java#L828-L836